pages/docs/reference/idioms.md

```diff
@@ -118,16 +118,17 @@ val list = listOf("a", "b", "c")
 
 <div class="sample" markdown="1" theme="idea" data-highlight-only>
 ```kotlin
-val map = mapOf("a" to 1, "b" to 2, "c" to 3)
+val map = mapOf("key-a" to "value-a", "key-b" to "value-b")
 ```
 </div>
 
 ### Accessing a map
 
 <div class="sample" markdown="1" theme="idea" data-highlight-only>
 ```kotlin
-println(map["key"])
-map["key"] = value
+val map = mutableMapOf("key-a" to "value-a", "key-b" to "value-b")
+map["key-c"] = "value-c"
+println(map["key-c"]) // prints "value-c"
```

Comment:
Not sure we should extend the idiom examples this much. This is just a short reminder, and the current example is pretty clear about what is key and what is value.
The complete runnable examples are provided in the [maps documentation](https://kotlinlang.org/docs/reference/map-operations.html).
I wouldn't change this part.

Reply:
I wanted to show here that you can not only change an entry, but also add one. Unfortunately, the mutableMapOf line is necessary for this, since mapOf from the previous section would not work.

I don't think Kotlin pros really need this page. Isn't it especially for newbies? And the more understandable the site, the fewer newbies will be put off by Kotlin.

Reply:
Well, in some sense this page _is helpful_ for newbies but it's _not intended for teaching_ Kotlin concepts. It just shows how common things look in Kotlin without explaining how they work and how/when to use them.
This is why we're keeping code samples here **as short as possible**, as opposed to samples [here](https://kotlinlang.org/docs/reference/collections-overview.html#map) for instance.
In your suggestion, the `map` declaration is valid but it's not reflecting what's written in the heading (**Accessing a map**). The other two lines are the same as the current ones.

Reply:
I wrote this in what I think is best. Ultimately, it is your decision to take it over, change it, or discard it. :-)